qwik/lib/vite.config.ts

```diff
@@ -0,0 +1,30 @@
+import { defineConfig } from 'vite';
+import pkg from './package.json';
+import { qwikVite } from '@builder.io/qwik/optimizer';
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+const { dependencies = {}, peerDependencies = {} } = pkg as any;
+const makeRegex = (dep) => new RegExp(`^${dep}(/.*)?$`);
+const excludeAll = (obj) => Object.keys(obj).map(makeRegex);
+
+export default defineConfig(() => {
+  return {
+    build: {
+      target: 'es2020',
+      lib: {
+        entry: './src/index.ts',
+        formats: ['es', 'cjs'],
+        fileName: (format) => `index.qwik.${format === 'es' ? 'mjs' : 'cjs'}`,
+      },
+      rollupOptions: {
```

Comment:
This could be updated to the latest version of our vite config for libraries (for example, with the "preserveModules" prop

Reply:
Updated